core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverManager.java

```diff
@@ -98,6 +99,65 @@ static <Solution_, ProblemId_> SolverManager<Solution_, ProblemId_> create(
         return new DefaultSolverManager<>(solverFactory, solverManagerConfig);
     }
 
+    // ************************************************************************
+    // Builder methods
+    // ************************************************************************
+
+    /**
+     * Creates a Builder that allows to customize and submit a planning problem to solve.
+     *
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     * @param problem never null, a {@link PlanningSolution} usually with uninitialized planning variables
+     * @return never null
+     */
+    default SolverJobFactory<Solution_, ProblemId_> builder(ProblemId_ problemId, Solution_ problem) {
+        return builder(problemId, (problemId_) -> problem);
+    }
+
+
+    /**
+     * As defined by {@link #builder(Object, Object)}
+     *
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     *        {@link #getSolverStatus(Object) to get the status} or if the problem changes while solving.
+     * @param problemFinder never null, function that returns a {@link PlanningSolution}, usually with uninitialized planning
+     *                      variables
+     * @return never null
+     */
+    SolverJobFactory<Solution_, ProblemId_> builder(ProblemId_ problemId,
+                                                    Function<? super ProblemId_, ? extends Solution_> problemFinder);
+
+    /**
+     * Creates a Builder that allows to customize and submit a planning problem to solve and listen to intermediate
+     * best solutions.
+     *
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     * @param problem never null, a {@link PlanningSolution} usually with uninitialized planning variables
+     * @param bestSolutionConsumer never null, called multiple times, on a consumer thread
+     * @return never null
+     */
+    default SolverAndListenerJobFactory<Solution_, ProblemId_> builder(ProblemId_ problemId, Solution_ problem,
+                                                               Consumer<? super Solution_> bestSolutionConsumer) {
+        return builder(problemId, (problemId_) -> problem, bestSolutionConsumer);
+    }
```

Comment:
I'd argue put `bestSolutionConsumer` on the builder, remove this overload.

Reply:
Let's wait for the design meeting to define the next steps here.

Reply:
Done!